src/main/java/com/google/jenkins/plugins/storage/AbstractUpload.java

```diff
@@ -270,6 +270,13 @@ public boolean forResult(Result result) {
   /** @return Provide detail information summarizing this download for the GCS upload report. */
   public abstract String getDetails();
 
+  /** @return The {@link UploadModule} for providing dependencies. */
+  protected UploadModule getModule() {
+    if (this.module == null) {
+      this.module = getDescriptor().getModule();
```

Comment:
if this operation isn't expensive, you could just return getDescriptor().getModule() instead and make this.module final again

Reply:
Done